packages/composites/anchor/src/endpoint/price/beth.ts

```diff
@@ -41,3 +25,33 @@ export const execute: PriceExecute = async (input, _, config, taAdapterResponse)
     },
   }
 }
+
+const getStETHExchangeRate = async (
+  config: Config,
+  provider: ethers.providers.JsonRpcProvider,
+): Promise<number> => {
+  const stEthPoolContract = new ethers.Contract(
+    config.stEthPoolContractAddress,
+    curvePoolAbi,
+    provider,
+  )
+  const ethBal = await stEthPoolContract.balances(0)
+  const stEthBal = await stEthPoolContract.balances(1)
+  return new BigNumber(ethBal.toString()).dividedBy(new BigNumber(stEthBal.toString())).toNumber()
+}
+
+const getStEthBEthExchangeRate = async (
+  config: Config,
+  provider: ethers.providers.JsonRpcProvider,
+): Promise<number> => {
+  const anchorVaultContract = new ethers.Contract(
+    config.anchorVaultContractAddress,
+    anchorVaultAbi,
+    provider,
+  )
+  const bEthExchangeRateBigNum = await anchorVaultContract.get_rate()
+  // Need to convert to BigNumber JS from ethers BigNumber as the latter will remove all decimals
+  return new BigNumber(bEthExchangeRateBigNum.toString())
+    .dividedBy(new BigNumber(10).pow(18))
+    .toNumber()
```

Comment:
Why are we converting this to a number, which would lose precision?

Reply:
So I turned this to a number as the result from token allocations is returned as number and they are multiplied together later on. Do you recommend converting the TA response to a big number first?

Reply:
decided to use BigNumber